tests/components/homee/conftest.py

```diff
@@ -0,0 +1,64 @@
+"""Fixtures for Homee integration tests."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from typing_extensions import Generator
+
+from homeassistant.components.homee.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+HOMEE_ID = "00055511EECC"
+HOMEE_IP = "192.168.1.11"
+TESTUSER = "testuser"
+TESTPASS = "testpass"
+
+
+@pytest.fixture
+def mock_config_entry() -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    return MockConfigEntry(
+        title=f"{HOMEE_ID} ({HOMEE_IP})",
+        domain=DOMAIN,
+        data={
+            CONF_HOST: HOMEE_IP,
+            CONF_USERNAME: TESTUSER,
+            CONF_PASSWORD: TESTPASS,
+        },
+        unique_id=HOMEE_ID,
+        version=1,
+        minor_version=1,
+    )
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Mock setting up a config entry."""
+    with patch(
+        "homeassistant.components.homee.async_setup_entry", return_value=True
+    ) as mock_setup:
+        yield mock_setup
+
+
+@pytest.fixture
+def mock_homee() -> Generator[MagicMock]:
+    """Return a mock Homee instance."""
+    with patch(
+        "homeassistant.components.homee.config_flow.Homee", autospec=True
+    ) as mocked_homee:
+        homee = mocked_homee.return_value
+
+        homee.host = HOMEE_IP
+        homee.user = TESTUSER
+        homee.password = TESTPASS
+        homee.settings = MagicMock()
+        homee.settings.uid = HOMEE_ID
+        homee.reconnect_interval = 10
+
+        homee.get_access_token.return_value = "test_token"
+        homee.wait_until_connected = AsyncMock()
+        homee.wait_until_disconnected = AsyncMock()
```

Comment:
if your mock is autospecced, I believe you can get rid of the mock definitions

Reply:
If I remove them:
```
FAILED tests/components/homee/test_config_flow.py::test_config_flow - TypeError: object MagicMock can't be used in 'await' expression
FAILED tests/components/homee/test_config_flow.py::test_flow_already_configured - TypeError: object MagicMock can't be used in 'await' expression
```

Reply:
Did you update the library with the async Def?

Reply:
```py

    def wait_until_connected(self):
        """Return a coroutine that runs until a connection has been established."""
        return self._connected_event.wait()

    def wait_until_disconnected(self):
        """Return a coroutine that runs until the connection has been closed."""
        return self._disconnected_event.wait()
```
it does not

Reply:
I tried, but that was in fact not the right thing to do. when I tried to make these functions async, weird things happened.
I'm not sure what is really correct here. The functions are sync, but return an async coroutine. Most interestingly this works, but when I tried to make the functions themselves async, it didn't work. Have to try it again, to give you details though...

Reply:
Oh right. Hmm. Interesting. Wouldn't this be a

```py
    async def wait_until_connected(self):
        """Return a coroutine that runs until a connection has been established."""
        return await self._connected_event.wait()
```
Would this make sense?

Reply:
Ahaha, I was just being stupid and forgot the `await` on the return statement... 🙄